app/volume.go

```diff
@@ -412,32 +439,6 @@ func (job *Job) getVolumeHeadSize(volume *longhornclient.Volume) (int64, error)
 	return volumeHeadSize, nil
 }
 
-func (job *Job) listSnapshotNamesForCleanup(snapshots []longhornclient.Snapshot) []string {
-	sts := []*NameWithTimestamp{}
-
-	// only remove snapshots that where created by our current job
-	jobLabel, found := job.labels[types.RecurringJobLabel]
-	if !found {
-		return []string{}
-	}
-	for _, snapshot := range snapshots {
-		snapshotLabel, found := snapshot.Labels[types.RecurringJobLabel]
-		if found && jobLabel == snapshotLabel {
-			t, err := time.Parse(time.RFC3339, snapshot.Created)
-			if err != nil {
-				logrus.Errorf("Fail to parse datetime %v for snapshot %v",
-					snapshot.Created, snapshot.Name)
-				continue
-			}
-			sts = append(sts, &NameWithTimestamp{
-				Name:      snapshot.Name,
-				Timestamp: t,
-			})
-		}
-	}
-	return job.getCleanupList(sts)
-}
-
 func (job *Job) getCleanupList(sts []*NameWithTimestamp) []string {
 	sort.Slice(sts, func(i, j int) bool {
 		if sts[i].Timestamp.Before(sts[j].Timestamp) {
```

Comment:
you can just do `return sts[i].Timestamp.Before(sts[j].Timestamp)`

Reply:
Thank you. The function is no longer used. So, I deleted it.